tapfreighter/chain_porter.go

```diff
@@ -880,14 +881,32 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 		// Submit the template PSBT to the wallet for funding.
 		//
 		// TODO(roasbeef): unlock the input UTXOs of things fail
-		feeRate, err := p.cfg.ChainBridge.EstimateFee(
-			ctx, tapscript.SendConfTarget,
+		var (
+			feeRate chainfee.SatPerKWeight
+			err     error
 		)
-		if err != nil {
-			return nil, fmt.Errorf("unable to estimate fee: %w",
-				err)
+
+		// First, use a manual fee rate if specified by the parcel.
+		// TODO(jhb): Support PSBT flow / PreSignedParcels
+		addrParcel, ok := currentPkg.Parcel.(*AddressParcel)
+		switch {
+		case ok && addrParcel.transferFeeRate != nil:
+			feeRate = *addrParcel.transferFeeRate
+			log.Infof("sending with manual fee rate")
+
+		default:
+			feeRate, err = p.cfg.ChainBridge.EstimateFee(
```

Comment:
Rather than conditionally call estimation here, we could instead lift it up to the rpcserver. So then we can drop the pointer to fee rate, as we compute that head of time.